### PR TITLE
Feature/LGA-3334 Case API Case Route

### PR DESCRIPTION
**Pull request by TawneeOwl**
## What does this pull request do?

Updates the case route to include all information.

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

[- [ ] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"](https://dsdmoj.atlassian.net/browse/LGA-3334)

### CODE REVIEW

File: app/models/cases.py [hunk 1]
```diff
@@ -38,6 +41,15 @@ class Case(BaseCase, TableModelMixin, table=True):
     )
 
 
+class CaseRetrieve(BaseCase, TableModelMixin):
```
Comment (said-moj) [inline app/models/cases.py:44]: Doesn't this return same/similar fields as the CaseResponse class?

File: app/models/cases.py [hunk 2]
```diff
@@ -26,6 +26,9 @@
 class BaseCase:
     case_type: CaseTypes = Field(index=True)
 
+    def retrieve(self, session: Session, instance_id: uuid.UUID) -> SQLModel | None:
```
Comment (said-moj) [inline app/models/cases.py:29]: There is already a retrieve method in the base request class https://github.com/ministryofjustice/laa-civil-case-api/blob/main/app/models/base.py#L82

File: app/routers/case_information.py [hunk 3]
```diff
@@ -18,11 +23,11 @@
 
 
 @router.get("/{case_id}", tags=["cases"])
-async def read_case(case_id: UUID, session: Session = Depends(get_session)) -> Case:
+async def read_case(case_id: UUID, session: Session = Depends(get_session)):
     case: Case | None = session.get(Case, case_id)
     if not case:
         raise HTTPException(status_code=404, detail="Case not found")
-    return case
+    return CaseResponse.from_orm(case)
```
Comment (said-moj) [inline app/routers/case_information.py:30]: What's the difference between returning `CaseResponse.from_orm(case)` and `case`

Comment (TawneeOwl) [reply app/routers/case_information.py:30]: From ORM reads the attributes of a class as defined by another class. It acts recursively https://docs.pydantic.dev/1.10/usage/models/#orm-mode-aka-arbitrary-class-instances

Comment (TawneeOwl) [reply app/routers/case_information.py:30]: If we just go for case it only returns the created at and id

**Review by said-moj — APPROVED** [review]

File: app/routers/case_information.py [hunk 4]
```diff
@@ -18,11 +23,11 @@
 
 
 @router.get("/{case_id}", tags=["cases"])
```
[the remark this answers is not in the document]
Comment (TawneeOwl) [reply app/routers/case_information.py]: Awesome find! I've updated

**Review by said-moj — APPROVED** [review]

File: app/routers/case_information.py [hunk 5]
```diff
@@ -17,8 +22,8 @@
 )
 
 
-@router.get("/{case_id}", tags=["cases"])
-async def read_case(case_id: UUID, session: Session = Depends(get_session)) -> Case:
+@router.get("/{case_id}", tags=["cases"], response_model=CaseResponse)
+async def read_case(case_id: UUID, session: Session = Depends(get_session)):
```
Comment (BenMillar-MOJ) [inline app/routers/case_information.py:26]: Could you re-add the `-> Case` docstring here please 👍

File: docs/source/documentation/case.html.md.erb [hunk 6]
```diff
@@ -119,7 +119,7 @@ You will receive the following response schema:
 GET /cases/
 ```
 
-### Read a case
+### Read all case information
```
Comment (BenMillar-MOJ) [inline docs/source/documentation/case.html.md.erb:122]: Could you update this to:

"Gets all case information for a given case id"

**Review by BenMillar-MOJ — CHANGES_REQUESTED** [review]

**Review by BenMillar-MOJ — APPROVED** [review]
LGTM

**Pull request merged**